Catappult/appcoins-wallet-android
# [task/APPC-3634] Compose with coroutines and result example

**What does this PR do?**

   Refactors the ChangeFiatCurrency feature to work with Coroutines and Result and implements the two screens in compose instead of view. 
   Includes the changes in the previous PR

**Database changed?**

   No

**Where should the reviewer start?**

- FiatCurrenciesRepository.kt

**How should this be manually tested?**

 - Open wallet and navigate to settings
 - Check the visible currency
 - Open the currency list
 - Change to a different currency

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-3634

**Code Review Checklist**
